src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -211,6 +212,18 @@ public function registerExtension(ExtensionInterface $extension)
         if (false !== $extension->getNamespace()) {
             $this->extensionsByNs[$extension->getNamespace()] = $extension;
         }
+
+        if ($extension instanceof CompilerPassProvidingExtensionInterface) {
```

Comment:
Following the previous suggestions, I would do this in `Kernel::prepareContainer()`, where the `build()` calls are happening. Likely inside the last extensions loop.

Reply:
I've instead moved this to the start of `ContainerBuilder::compile()`. The idea of this feature is to do this within the DI component, as it's useful for standalone use-cases of the DI component. When using it with HttpKernel, I would always recommend to use `Kernel::build()` or `BundleInterface::build()` instead.